src/server/middleware/log/log.go

```diff
@@ -40,6 +45,61 @@ func Middleware() func(http.Handler) http.Handler {
 			r = r.WithContext(ctx)
 		}
 
-		next.ServeHTTP(w, r)
+		e := &commonevent.Metadata{
+			Ctx:           r.Context(),
+			Username:      "unknown",
+			RequestMethod: r.Method,
+			RequestURL:    r.URL.String(),
+		}
+		if matched, resName := e.PreCheck(); matched {
```

Comment:
we can leverage the  middleware.BeforeRequest and  middleware.AfterRequest to make it more readable.

Reply:
For user login event, it requires to parse the request content to get the username, use the BeforeRequest and AfterRequest can't read the request body.